packages/esbuild-plugin-import-path-remapper/src/index.ts

```diff
@@ -0,0 +1,88 @@
+import { Plugin } from "esbuild";
+
+const resolvePath = (path: string, resolveDir: string): string | undefined => {
+  try {
+    return require.resolve(path, { paths: [resolveDir] });
+  } catch {
+    return undefined;
+  }
+};
+
+const findMainSourceFile = (
+  sourcePath: string,
+  resolveDir: string
+): string | undefined => {
+  const { main } = require(require.resolve(`${sourcePath}/package.json`, {
+    paths: [resolveDir],
+  }));
+
+  if (!main) {
+    return;
+  }
+
+  const remappedPath = `${sourcePath}/${main.replace(
+    /^(?:\.\/)?lib\/(.*)\.js/,
```

Comment:
Optionally match `x` as well:
```suggestion
    /^(?:\.\/)?lib\/(.*)\.jsx?/,
```

Reply:
I'm going to pretend I didn't copy/paste this from your babel code 😊. I should refactor this out to a shared package like Jason suggests.